libiscsiusr/tests/test_session.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Gris Ge <fge@redhat.com>
+ */
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <string.h>
+#include <pthread.h>
+#include <unistd.h>
+#include <assert.h>
+
+#include <libiscsiusr/libiscsiusr.h>
+
+#define _assert_print_prop_str_can_empty(struct_name, obj, prop_name) \
+	do { \
+		assert(struct_name##_##prop_name##_get(obj) != NULL); \
+		printf("\t" # prop_name ": '%s'\n", \
+		       struct_name##_##prop_name##_get(obj)); \
+	} while(0)
+
+#define _assert_print_prop_str_not_empty(struct_name, obj, prop_name) \
+	do { \
+		assert(struct_name##_##prop_name##_get(obj) != NULL); \
+		assert(strlen(struct_name##_##prop_name##_get(obj)) != 0); \
+		printf("\t" # prop_name ": '%s'\n", \
+		       struct_name##_##prop_name##_get(obj)); \
+	} while(0)
+
+#define _assert_print_prop_u32_not_zero(struct_name, obj, prop_name) \
+	do { \
+		assert(struct_name##_##prop_name##_get(obj) != 0); \
+		printf("\t" # prop_name ": %" PRIu32 "\n", \
+		       struct_name##_##prop_name##_get(obj)); \
+	} while(0)
+
+#define _assert_print_prop_i32_not_zero(struct_name, obj, prop_name) \
+	do { \
+		assert(struct_name##_##prop_name##_get(obj) != 0); \
+		printf("\t" # prop_name ": %" PRIi32 "\n", \
+		       struct_name##_##prop_name##_get(obj)); \
+	} while(0)
+
+static void test_session(struct iscsi_session *se);
+
```

Comment:
Why declare this right before defining it?

Reply:
Just a old habit of `declare then define`. I could remove these lines if that's required.